ddtrace/internal/glob_matching.py

```diff
@@ -0,0 +1,50 @@
+from .utils.cache import cachedmethod
+
+
+class GlobMatcher:
+    """This is a backtracking implementation of the glob matching algorithm.
+    The glob pattern language supports `*` as a multiple character wildcard which includes matches on `""`
+    and `?` as a single character wildcard, but no escape sequences.
```

Comment:
A usage example might be nice here, and probably worth calling out the cache logic since one might wonder why this is a class instead of just a function